src/Compilers/Core/Portable/PEWriter/FullMetadataWriter.cs

```diff
@@ -81,9 +84,11 @@ internal sealed class FullMetadataWriter : MetadataWriter
             bool metadataOnly,
             bool deterministic,
             bool emitTestCoverageData,
+            Encoding? defaultSourceEncoding,
+            Encoding? fallbackSourceEncoding,
```

Comment:
Rather then passing the encodings here, I think it'd be better to make them available on CommonPEModuleBuilder for consistency with other emit options.
See [http://sourceroslyn.io/#Microsoft.CodeAnalysis/Emit/CommonPEModuleBuilder.cs,66](http://sourceroslyn.io/#Microsoft.CodeAnalysis/Emit/CommonPEModuleBuilder.cs,66)

Reply:
Maybe we could just store the EditOptions reference rather than pulling each option out of them.

---
In reply to: [434109985](https://github.com/dotnet/roslyn/pull/44373#discussion_r434109985) [](ancestors = 434109985)

Reply:
Surface EmitOptions in CommonPEModuleBuilder